src/java.base/share/classes/jdk/internal/jrtfs/ExplodedImage.java

```diff
@@ -254,19 +255,20 @@ private void initNodes() throws IOException {
                     String moduleName = module.getFileName().toString();
                     // make sure "/modules/<moduleName>" is created
                     findModulesNode(MODULES + moduleName);
-                    Files.walk(module).filter(Files::isDirectory).forEach((p) -> {
-                        p = module.relativize(p);
-                        String pkgName = slashesToDots(p.toString());
-                        // skip META-INFO and empty strings
-                        if (!pkgName.isEmpty() && !pkgName.startsWith("META-INF")) {
-                            List<String> moduleNames = packageToModules.get(pkgName);
-                            if (moduleNames == null) {
-                                moduleNames = new ArrayList<>();
-                                packageToModules.put(pkgName, moduleNames);
+                    try (Stream<Path> contentsStream = Files.walk(module)) {
+                        contentsStream.filter(Files::isDirectory).forEach((p) -> {
+                            p = module.relativize(p);
+                            String pkgName = slashesToDots(p.toString());
+                            // skip META-INFO and empty strings
+                            if (!pkgName.isEmpty() && !pkgName.startsWith("META-INF")) {
+                                List<String> moduleNames = packageToModules.get(pkgName);
+                                if (moduleNames == null) {
+                                    moduleNames = new ArrayList<>();
+                                    packageToModules.put(pkgName, moduleNames);
+                                }
+                                moduleNames.add(moduleName);
                             }
-                            moduleNames.add(moduleName);
-                        }
-                    });
+                        });
```

Comment:
There is a missing closing curly brace here (which should come on the next line), but otherwise looks fine.